src/Symfony/Component/Messenger/WorkerMetadata.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger;
+
+/**
+ * @author Oleg Krasavin <okwinza@gmail.com>
+ */
+final class WorkerMetadata
+{
+    private $metadata;
+
+    public function __construct(array $metadata)
+    {
+        $this->metadata = $metadata;
+    }
+
+    public function set(string $key, $value): void
```

Comment:
Since populating the object is purely an internal operation, I don't think it makes sense to pollute the class with a bunch of individual setters for each property here. Hence the generic `set` method.